### DISCUSSION TITLE
PY請教

### DISCUSSION THREAD

**Discussion by z3107wii** [NanoVNA-Saver/nanovna-saver#636] [General]
您好，我是個PY初學者，想詢問您此軟體是用什麼去編譯的?
下載您的代碼後該如何修改後再做編譯讓他成為exe檔案的方式呢? 如果您願意花時間回復我我會很感謝您的，謝謝!

**Comment by Yannick-R59** [comment]
You need to have a nanovna. This is a electrical measurement tool that could be connect to a PC. So this software allows to control the tool and do measurements. Probably not the most useful way to learn python if you have not a nanovna.

If you have, you can probably download version 5.5 (the last that use QT5) of this software which was in the past cross-platform and could be compiled and run on windows or Debian or other environnement. The last version could only be compiled on specific and rare system that use QT6.

If you use the old one and windows, you can use for example anaconda and spyder to have python ide, open nanovnasaver.py and run. That will compile and be executed.

having the code allow you to know what do exactly the software and eventually change it to fit your need.
